src/metabase/search/engines/appdb/specialization/postgres.clj

```diff
@@ -0,0 +1,174 @@
+(ns metabase.search.engines.appdb.specialization.postgres
+  (:require
+   [clojure.string :as str]
+   [metabase.search.engines.appdb.specialization :as specialization]
+   [metabase.util :as u]
+   [toucan2.core :as t2]))
+
+(def ^:private tsv-language "english")
+
+(defmethod specialization/table-schema :postgres []
+  [[:id :bigint [:primary-key] [:raw "GENERATED BY DEFAULT AS IDENTITY"]]
+   ;; entity
+   [:model_id :int :not-null]
+   [:model [:varchar 254] :not-null]                        ;; TODO We could shrink this to just what we need.
+   [:name :text :not-null]
+   ;; search
+   [:search_vector :tsvector :not-null]
+   [:with_native_query_vector :tsvector :not-null]
+   ;; results
+   [:display_data :text :not-null]
+   [:legacy_input :text :not-null]
+   ;; scoring related
+   [:dashboardcard_count :int]
+   [:last_viewed_at :timestamp-with-time-zone]
+   [:official_collection :boolean]
+   [:pinned :boolean]
+   [:verified :boolean]
+   [:view_count :int]
+   ;; permission related entities
+   [:collection_id :int]
+   [:database_id :int]
+   ;; filter related
+   [:archived :boolean :not-null [:default false]]
+   [:creator_id :int]
+   [:last_edited_at :timestamp-with-time-zone]
+   [:last_editor_id :int]
+   [:model_created_at :timestamp-with-time-zone]
+   [:model_updated_at :timestamp-with-time-zone]
+   ;; useful for tracking the speed and age of the index
+   [:created_at :timestamp-with-time-zone
+    [:default [:raw "CURRENT_TIMESTAMP"]]
+    :not-null]
+   [:updated_at :timestamp-with-time-zone :not-null]])
+
+;; TODO I strongly suspect that there are more indexes that would help performance, we should examine EXPLAIN.
```

Comment:
Cost without an index is `0..584` for me right now (stats, but locally), and `0..68` with one:

```sql
create index search_index_gin on search_index (collection_id, database_id, table_id, creator_id, archived);
```